R/map_layer_h3_hexagon.R

```diff
@@ -66,6 +66,22 @@ mapdeckH3JSDependency <- function() {
 #' key <- 'abc'
 #' set_token( key )
 #'
+#' # TODO: Move to file somewhere.
+#' road_safety <- read_csv("https://raw.githubusercontent.com/uber-common/deck.gl-data/master/examples/3d-heatmap/heatmap-data.csv")
+#' road_safety_h3 <- road_safety %>%
+#' 	transmute(h3 = h3js::h3_geo_to_h3(lat, lng, 7)) %>%
```

Comment:
is `transmute` a dplyr verb?
the examples have to be self-contained within the package (and its dependencies). `dplyr` is not a dependency, and neither is `h3js`.

I suggest creating a small data set, similar to what I've done [here](https://github.com/SymbolixAU/mapdeck/tree/master/data-raw), then use `usethis::use_data( mydata )` to save it.

You'll then need to [document it](https://github.com/SymbolixAU/mapdeck/blob/master/R/data.R)

Reply:
Yep - this is WIP.
I had trouble installing on my new laptop and wanted to push to my branch so I could continue working on it there. Except I had more trouble getting it working on the old laptop.

The data preparation is planned to be 'moved somewhere' (either with `use_data()`, or in the github page to reduce package bloat). This will remove dependencies on h3js, readr, and dplyr. Hence the TODO.

```
 unable to load shared object 'mapdeck.so':
  mapdeck.so: undefined symbol: _ZNK5boost9gregorian10greg_month15as_short_stringEv
```

If by any chance you know how to fix this issue (somewhere from boost I imagine), then that would help finalise this PR.

Reply:
I see.

looks like you maybe need to install `library(BH)`?

Reply:
``` r
packageVersion("BH")
#> [1] '1.72.0.3'
```

<sup>Created on 2020-01-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
So I've finally figured out a fix for this.
Adding `PKG_LIBS = -L/usr/lib -lboost_date_time` to the Makevars seems to have done the trick.

In the process I also installed the latest version of boost from sources, but I don't think this made a notable difference.

Reply:
Are you using `gcc` compiler by any chance?

Reply:
as of v0.3.1 (now on CRAN) I've removed `boost_date_time` from all dependencies, so this option should no longer be needed

Reply:
Sounds great, thanks!